### ISSUE TITLE
Question about relay and SSL/TLS

### ISSUE THREAD

**Issue by blackholegalaxy** [tomav/docker-mailserver#957]
We are interested in using relay mode for docker-mailserver. A question though which is not documented in the relay wiki page.

Our external SMTP requires the use to SSL or TLS. How to setup the relay for these modes to be used?  Will docker-mailserver guess and auto apply either mode depending on RELAY_PORT value?

Then do we have to send mail to docker-mailserver on the same as RELAY_PORT or on 25?

**Comment by johansmitsnl** [comment]
The wiki https://github.com/tomav/docker-mailserver/wiki/Configure-Relay-Hosts uses TLS

**Comment by blackholegalaxy** [comment]
Yes but what is the port I need to send my mail against in order for the email to be relayed to my server using TLS?

I saw on your wiki we can have `@domain1.com        [relay1.org]:587`  but it does not stats what is the port we have to use to send the email to docker-mailserver.

Do we have to do:

`backend --> docker-mailserver:25 --> relay:587`

Or:

`backend --> docker-mailserver:587 --> relay:587`

**Comment by crazystick** [comment]
TLDR: the second one, assuming "backend" is a mail client, not another mailserver.

docker-mailserver should use TLS if it's offered by your relay, see http://www.postfix.org/postconf.5.html#smtp_use_tls which is set to yes when you add a relay host.

docker-mailserver will only accept connections from mail clients on 465 and 587 as per the README.

**Comment by blackholegalaxy** [comment]
Our backend is indeed a java application acting as mail client to send email through SMTP. So we were looking for proper settings to put on our code to send emails through docker-mailserver (which will act as relay to the real SMTP given by our customer). 

Your explanation makes things clearer for us. Another question comes in mind though: our java backend (client) will send the email on 587. But I am right saying my backend doesn't have to know the relay user and password?

We basically just have to send email to docker-mailserver on port 587, then, docker-mailserver will relay the email with proper user and password? So our code doesn't have to know about these credentials.

**Comment by crazystick** [comment]
Right - your backend doesn't know anything about the relay. The relay info lives in the configuration of docker-mailserver.

**Comment by crazystick** [comment]
@blackholegalaxy is this all good now? can it be closed?

**Comment by blackholegalaxy** [comment]
Yes of course thanks for explanations.
